src/bloomf.ml

```diff
@@ -78,6 +78,50 @@ let size_estimate t =
   let xf = float_of_int (Bitv.pop t.b) in
   int_of_float (-.mf /. kf *. log (1. -. (xf /. mf)))
 
+let to_bytes t =
+  let flat_p_len =
+    t.p_len |> List.fold_left (fun acc (a, b) -> a :: b :: acc) []
+  in
+  let first_to_encode = Array.of_list (t.m :: t.k :: flat_p_len) in
+  Bytes.concat Bytes.empty
+    [
+      Bytes.init
+        (8 * Array.length first_to_encode)
+        (fun i -> Char.chr ((first_to_encode.(i / 8) lsr (8 * i)) land 0xFF));
+      Bitv.to_bytes t.b;
+    ]
+
+let of_bytes b =
+  let int_of_bytes b =
+    let rec build x i =
+      if i < 0 then x
+      else build ((x lsl 8) lor Char.code (Bytes.get b i)) (pred i)
+    in
+    build 0 7
+  in
+  let rec parse_p_len start stop acc last =
+    if start >= stop then (start, acc)
+    else
+      let next_value = int_of_bytes (Bytes.sub b start 8) in
+      match last with
+      | Some value ->
+          parse_p_len (start + 8) stop ((value, next_value) :: acc) None
+      | None -> parse_p_len (start + 8) stop acc (Some next_value)
+  in
+  match int_of_bytes (Bytes.sub b 0 8) with
```

Comment:
You assume here that the `integer` is a `int64` (8 bytes) but it does not work on 32-bits architecture. To solve that, you should use `Int64` and make something like the `int64_of_bytes` function.

Reply:
(May be for more context, an OCaml `int` on a 32 bits architecture is a 31-bits integer - on a 64-bits architecture, it's a 63-bits integer - you can easily simulate the situation with an OCaml switch `4.10.0+32bit` - `opam switch create 4.10.0+32bit`)

Reply:
I knew the bit size of native integers was smaller on 32 bit systems, but the issue I see with this is that the x86_32 builds passed, which seems odd to me. I'll check this out on a 32bit switch (I've not had a lot of time for this in the recent weeks, but I should be able to test this in the next few days). 

Thank you for the review.